### PR TITLE
Static analysis for non-OO aspects

### PR DESCRIPTION
**Pull request by hinescd**

### CODE REVIEW

File: src/main/java/com/github/lessjava/visitor/impl/LJStaticAnalysis.java [hunk 1]
```diff
@@ -68,9 +81,18 @@ public void preVisit(ASTAssignment node) {
...
 
+    /**
+     * Make sure we don't assign to some reserved words
+     *
+     * @param node an assignment node
+     */
     @Override
     public void postVisit(ASTAssignment node) {
         super.postVisit(node);
+        List<String> reservedWords = List.of("this", "super", "class", "int", "boolean", "double", "char", "long", "float");
```
Comment (Zamua) [inline src/main/java/com/github/lessjava/visitor/impl/LJStaticAnalysis.java]: Perhaps reservedWords should be a final static variable of the class

**Comment by Zamua** [comment]
Some of the tests from `./dotests.sh -r tests` also failed. Is that concerning?

File: src/main/java/com/github/lessjava/visitor/impl/LJStaticAnalysis.java (hunk 1 above)
Comment (hinescd) [reply src/main/java/com/github/lessjava/visitor/impl/LJStaticAnalysis.java]: Good idea, addressed

**Comment by hinescd** [comment]
Yeah, some of those will fail until my portion of the project is done. If you're not comfortable with this being in master yet, we can keep it in a different branch until they're passing. I don't want to submit my whole project at once though, I'm sure that would be a nightmare to review.

**Comment by Zamua** [comment]
No this is fine for now. Future projects should probably be done how you described. A different branch until complete and then merged into master. I'll give this more thought in the future 

🤷‍♀️

**Pull request merged**